app/uk/gov/hmrc/gform/builder/BuilderController.scala

```diff
@@ -632,4 +633,147 @@ class BuilderController(
     allFormComponentIds diff visibleFormComponentIds
   }
 
+  def originalAcknowledgement(formTemplateId: FormTemplateId) =
+    auth.authAndRetrieveForm[SectionSelectorType.Normal](
+      formTemplateId,
+      None,
+      OperationWithForm.EditFormAcknowledgement
+    ) { implicit request => lang => cache => sse => formModelOptics =>
+      gformConnector.getFormTemplateRaw(formTemplateId).flatMap { formTemplateRaw =>
+        val jsonString: String = PlayJson.stringify(formTemplateRaw)
+        val maybeCirceJson: Either[ParsingFailure, Json] =
+          io.circe.parser.parse(jsonString)
+
+        val json: Option[Json] = maybeCirceJson.toOption.flatMap { json =>
+          json.hcursor.downField("acknowledgementSection").focus
+        }
+        json match {
+          case None                         => InternalServerError("Unable to retrieve acknowledgement section").pure[Future]
+          case Some(acknowledgementSection) => Ok(acknowledgementSection).pure[Future]
+        }
+      }
+
+    }
+
+  def generateAcknowledgementPanelHtml(formTemplateId: FormTemplateId) =
+    auth.authAndRetrieveForm[SectionSelectorType.Normal](
+      formTemplateId,
+      None,
+      OperationWithForm.EditFormAcknowledgement
+    ) { implicit request => _ => cache => implicit sse => formModelOptics =>
+      import i18nSupport._
+      val formTemplate = cache.formTemplate
+      val envelopeId = cache.form.envelopeId
+
+      val destinationList = formTemplate.destinations match {
+        case destinationList: DestinationList => destinationList
+        case _                                => throw new Exception("")
```

Comment:
Throwing exception will not display as an error on the UI.
Please use `badRequest("Please explain what went wrong here").pure[Future]`
Also change in builder is needed for this to work.
The change there will be something like
```
const error = `Error when rendering acknowledgement panel html. ${json.error}`;
```
Instead of
```
const error = `Error when rendering acknowledgement panel html. Body content: ${json}`;
```